### PR TITLE
feat: get all blogs with pagination

### PR DESCRIPTION
**Pull request by ObodoakorDavid**
## What does the PR do?

This PR is about adding the endpoint for retrieving all blogs with pagination

How should this be manually tested?
Send a GET request to /api/v1/blogs (This will automatically default the params of page and page_size to 1 and10 respectively)

Alternatively, you can provide your own values by sending a GET request to /api/v1/blogs?page=2&page_size=5

Request Body

```

No Request Body Required

```

200 Successful Response:

```

{
    "status": 200,
    "message": "Blogs retrieved successfully",
    "status_code": 200,
    "count": 2,
    "next": "http://example.com/api/v1/blogs?page=11&page_size=10",
    "previous": null,
    "data": [
        {
            "blog_id": "c6947b63-4ef3-4a4d-ae9a-82d481878d5b",
            "title": "Learn NestJS In One Week",
            "content": "Unprecedented shege awaits",
            "image_urls": [
                "https://example.com/image1.jpg",
                "https://example.com/image2.jpg"
            ],
            "tags": [
                "NestJS",
                "TypeScript",
                "Backend"
            ],
            "author": "John Doe",
            "created_at": "2024-08-05T18:34:10.426Z"
        },
        {
            "blog_id": "307096ae-5f55-4bfe-8351-03b1ebac4dff",
            "title": "5 Things you need to know before doing HNG",
            "content": "You will cry",
            "image_urls": [
                "https://example.com/image1.jpg",
                "https://example.com/image2.jpg"
            ],
            "tags": [
                "Shege",
                "Tears",
            ],
            "author": "John Doe",
            "created_at": "2024-08-05T18:44:32.107Z"
        }
    ],
    "page": "1",
    "page_size": "2"
}

```

Error Responses: 

400 Bad Request:

```
{
    "message": "Page number must be a positive integer",
    "status_code": 400
}
```

#### Checklist of what id did

- Implemented an endpoint to get all blogs
- Implemented pagination with URL params
- Added unit tests for the blog service. 

Reference Issue
[FEAT](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/94) Get All Blog Posts With Pagination

### CODE REVIEW

File: src/modules/blogs/blogs.service.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        status_code: 200,
+        message: 'Blogs retrieved successfully',
+        count: total,
+        next: nextLink,
+        previous,
+        data: blogs,
+        page,
+        page_size,
+      };
+    } catch (error) {
```
Comment (Cobidex) [inline src/modules/blogs/blogs.service.ts:66]: no need for try catch. there is an interceptor for non http exception

**Comment by TheCodeGhinux** [comment]
fix conflicts

**Comment by ObodoakorDavid** [comment]
> fix conflicts

Done. Please review.

**Comment by ObodoakorDavid** [comment]
> fix conflicts

Fixed.

File: src/modules/blogs/blogs.service.ts [hunk 2]
```diff
@@ -51,6 +51,38 @@ export class BlogService {
     };
   }
 
+  async findAll(page = 1, page_size = 10) {
+    if (!Number.isInteger(page) || page < 1) {
+      throw new BadRequestException('Page number must be a positive integer');
+    }
```
Comment (davidolanrewaju) [inline src/modules/blogs/blogs.service.ts:57]: Please use the global exception to throw your errors

File: src/modules/blogs/blogs.controller.ts [hunk 3]
```diff
@@ -39,4 +40,14 @@ export class BlogController {
...
+
+  @skipAuth()
+  @ApiOperation({ summary: 'Get All Blogs' })
+  @ApiResponse({ status: 200, description: 'Blogs retrieved successfully', type: BlogResponseDto })
+  @ApiResponse({ status: 400, description: 'Page or Page_size must be a positive number' })
+  @Get()
+  async findAll(@Query('page') page: number = 1, @Query('page_size') page_size: number = 10) {
+    const result = await this.blogService.findAll(page, page_size);
+    return result;
+  }
```
Comment (CFCIfe) [inline src/modules/blogs/blogs.controller.ts:52]: Use descriptive name

```suggestion
async getAllBlogs(@Query('page') page: number = 1, @Query('page_size') page_size: number = 10) {
    const result = await this.blogService.getAllBlogs(page, page_size);
    return result;
  }
```

**Review by davidolanrewaju — DISMISSED** [review]

**Pull request closed without merging**